phpBB/phpbb/routing/router.php

```diff
@@ -160,10 +170,118 @@ public function find($base_path = '')
 			}
 		}
 
+		$this->resolveParameters($this->route_collection);
+
 		return $this;
 	}
 
 	/**
+	 * Replaces placeholders with service container parameter values in:
+	 * - the route defaults,
+	 * - the route requirements,
+	 * - the route path,
+	 * - the route host,
+	 * - the route schemes,
+	 * - the route methods.
+	 *
+	 * @param RouteCollection $collection
+	 */
+	private function resolveParameters(RouteCollection $collection)
+	{
+		foreach ($collection as $route)
+		{
+			foreach ($route->getDefaults() as $name => $value)
+			{
+				$route->setDefault($name, $this->resolve($value));
+			}
+
+			foreach ($route->getRequirements() as $name => $value)
+			{
+				if ($name === '_scheme' || $name === '_method')
+				{
+					continue; // ignore deprecated requirements to not trigger deprecation warnings
```

Comment:
Unset using the key once instead of checking over and over again?
```php
$requirements = $route->getRequirements();
unset($requirements['_scheme']);
unset($requirements['_method']);
foreach ($requirements)
{
    $route->setRequirement($name, $this->resolve($value));
}
```